File: DFS&BFS/javascript/2667.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);

let graph = [];
for (let i = 1 ; i<input.length-1 ; i++){
  graph.push([]);
  for (let j = 0 ; j<input[i].length ; j++){
    graph[i-1].push(Number(input[i][j]));
  }
}
let result = [];
let cnt,houseCnt= 0;
const moves = [[-1,0],[0,1],[1,0],[0,-1]];
const rangeCheck = (x,y) => {
  if (x>=n || y>= n || x<0 || y<0) return false;
  return true;
}
const dfs = (r,c) => {
  if (graph[r][c] === 0) return 0;
  graph[r][c] = 0;
  let nx,ny;
  for (let move of moves){
    nx = r + move[0];
    ny = c + move[1];
    if (rangeCheck(nx,ny) && graph[nx][ny] === 1) {
      cnt++;
      dfs(nx,ny);
    }
  }
  return 1;
}

for (let i = 0 ; i<n ; i++){
  for (let j = 0 ; j<n ; j++){
    if (graph[i][j] === 1){
      cnt = 1;
      dfs(i,j);
      result.push(cnt);
      houseCnt++;
    }
  }
}
console.log(houseCnt);
console.log(result.sort((a,b)=>a-b).join("\n"));

File: DFS&BFS/python/2667.py
from collections import deque
n = int(input())
graph = []
houses = []
house_cnt = 0
for i in range(n):
    graph.append(list(map(int,input())))

def bfs(x,y):
    if graph[x][y] == 0:
        return False
    queue = deque()
    queue.append((x,y))
    dx = [-1,1,0,0]
    dy = [0,0,-1,1]
    graph[x][y] = 0
    result = 1 
    while queue:
        x,y = queue.popleft()
        for i in range(len(dx)):
            nx = x+dx[i]
            ny = y+dy[i]
            if nx>=0 and ny>= 0 and nx<n and ny<n and graph[nx][ny] == 1:
                graph[nx][ny] = 0
                queue.append((nx,ny))
                result += 1
    return result
for i in range(n):
    for j in range(n):
        if graph[i][j] == 0 : continue
        houses.append(bfs(i,j))
        house_cnt+=1
houses.sort()
print(house_cnt)
for house in houses:
    print(house)